sparkly/functions.py

```diff
@@ -153,3 +153,23 @@ def _execute_case(accumulator, case):
     result = reduce(_execute_case, cases, F).otherwise(default)
 
     return result
+
+
+def argmax(field, nullable, *by, condition=None):
+    """Get t he `field` value that maximizes columns in `by`
+
+    Arguments:
+        field (string): the field to return (and the alias to use)
+        nullable (boolean): whether the field can be None
+        by (*string): fields to maximize. In reality, this will usually be
+            only one field. But you may use multiple for tiebreakers
+        condition (optional): Only consider the entities that pass this condition,
+    """
+    fields = by + (field,)
+    result = F.struct(*fields)
+    if condition is not None:
+        result = F.when(condition, result)
+    if not nullable:
```

Comment:
doesn't seem like the `nullable` param is necessary.  you could just do `argmax ..., condition=F.col(field).isNotNull())`, right?

Reply:
nevermind, Edmund and I talked offline. He will make `nullable` an optional param that defaults to `False`, since we that null filter in most of our current use-cases